docs/vectorizer/document-vectorization.md

```diff
@@ -0,0 +1,516 @@
+
+#### Grant Permissions to your bucket to the role
+
+```bash
+aws iam put-role-policy \
+  --role-name timescale-vectorizer-s3-access \
+  --policy-name S3AccessPolicy \
+  --policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Action": [
+          "s3:GetObject"
+        ],
+        "Effect": "Allow",
+        "Resource": [
+          "arn:aws:s3:::test",
+          "arn:aws:s3:::test/*"
+        ]
+      }
+    ]
+  }'
+```
+
+#### Get the role ARN
+```bash
+aws iam get-role --role-name timescale-s3-role-test --query 'Role.Arn' --output text
+```
+#### Configure it in your ai.loading_uri:
+
+```sql
+ai.loading_uri(
+    column_name => 'uri',
+    aws_role_arn => 'arn:aws:iam::123456789012:role/timescale-vectorizer-s3-access'
+)
+```
+
+
+### Syncing S3 to a Documents Table
+
+If your application so far does not handle document uploads which would allow you to update the document table directly. You can instead use s3 events to keep your document table synchronized with S3 when documents are uploaded, modified, or deleted:
+
+```python
+# AWS Lambda function to handle S3 events
+def handle_s3_event(event, context):
+    import psycopg2
+    
+    # Extract document info from S3 event
+    bucket = event['Records'][0]['s3']['bucket']['name']
+    key = event['Records'][0]['s3']['object']['key']
+    event_name = event['Records'][0]['eventName']
+    
+    # Connect to PostgreSQL
+    conn = psycopg2.connect("postgresql://user:password@host:port/database")
+    cursor = conn.cursor()
+    
+    # Handle different event types
+    if 'ObjectCreated' in event_name:
+        # Document created or updated
+        cursor.execute(
+            """
+            INSERT INTO document (title, uri, updated_at) 
+            VALUES (%s, %s, CURRENT_TIMESTAMP)
+            ON CONFLICT (uri) DO UPDATE 
+            SET updated_at = CURRENT_TIMESTAMP
+            """,
+            [key.split('/')[-1], f"s3://{bucket}/{key}"]
+        )
+    elif 'ObjectRemoved' in event_name:
+        # Document deleted
+        cursor.execute(
+            "DELETE FROM document WHERE uri = %s",
+            [f"s3://{bucket}/{key}"]
+        )
+    
+    conn.commit()
+    cursor.close()
+    conn.close()
+```
+
+Configure an [S3 bucket notification](https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventNotifications.html) to trigger this Lambda function on object events (PutObject, DeleteObject).
+
+
+### Common Issues and Solutions
+
+**1. S3 Access Issues**
+
+If documents from S3 fail to load:
+- Verify AWS credentials are correctly configured
+- Check that IAM roles have appropriate permissions
+- Ensure S3 bucket names and object keys are correct
+
+
+**2. Embedding API Rate Limits**
```

Comment:
```suggestion
**2. Embedding services API Rate Limits**
```

Or similar. Whatever it clarifies this is out of our responsibility, not a pgai thing.